src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -652,7 +652,7 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
             _incomingBuffer.Discard(frameHeader.Length);
         }
 
-        private async Task StartWriteAsync(int writeBytes, CancellationToken cancellationToken = default)
+        private async Task<Memory<byte>> StartWriteAsync(int writeBytes, CancellationToken cancellationToken = default)
```

Comment:
`ValueTask<Memory<byte>>`

Reply:
@stephentoub Can you please specify the reason behind using ValueTask over here?

Reply:
This will very frequently complete synchronously, on a hot path, and so it's better to avoid the task allocation.  Previously in such a case this wouldn't allocate because a completed task singleton would be returned, but the runtime can't possible cache a `Task<Memory<byte>>` for every possible `Memory<byte>`, so this will allocate a new task for every invocation.  `ValueTask<Memory<byte>>` avoids that.